simplq/src/api/useRequest.js

```diff
@@ -0,0 +1,48 @@
+import { useAuth0 } from '@auth0/auth0-react';
+import { useCallback } from 'react';
+import makeRequest from './axios';
+
+// Use this hook to get a function that can be used to make
+// authenticated API calls from inside functional components.
+// Example:
+//
+// import { QueueRequestFactory } from 'api/requestFactory';  // Request structure is defined here.
+//
+// const MyAwesomeComponent = () => {
+//   const { requestMaker } = useRequest();
+//   ...
+//   requestMaker(QueueRequestFactory.create(textFieldValue))
+//     .then((response) => {
+//       // Handle response here
+//       ...
+//     });
+// }
+
```

Comment:
Kudos on the description

Reply:
I feel a bit more commenting is required in the request factory side. Let me add